st2common/st2common/services/packs.py

```diff
@@ -290,3 +291,95 @@ def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
             'The action metadata file "%s" does not exists on disk.',
             action_metadata_file_path,
         )
+
+
+def clone_content_to_destination_file(source_file, destination_file):
+    try:
+        with open(source_file, "r") as sf:
+            with open(destination_file, "w") as df:
+                for line in sf:
+                    df.write(line)
+    except PermissionError:
+        LOG.error(
+            'No write permission for "%s" file',
+            destination_file,
+        )
+        msg = 'No permission to write in "%s" file' % (destination_file)
+        raise PermissionError(msg)
+    except FileNotFoundError:
+        LOG.error('No "workflows" directory at path: "%s"', destination_file)
+        msg = 'Please make sure "workflows" directory present at path: "%s"' % (
+            destination_file
+        )
+        raise FileNotFoundError(msg)
+    except Exception as e:
+        LOG.error(
+            'Could not copy to "%s" file. Exception was "%s"',
+            destination_file,
+            e,
+        )
+        msg = (
+            'Could not copy to "%s" file, please check the logs '
+            "or ask your StackStorm administrator to check and clone "
+            "the actions files manually" % (destination_file)
+        )
+        raise Exception(msg)
+
+
+def clone_action(
+    source_pack_base_path,
+    source_metadata_file,
+    source_entry_point,
+    dest_pack_base_path,
+    dest_pack,
+    dest_action,
+):
+    """
+    Prepares the path for entry point and metadata files for source and destination.
+    Clones the content from source action files to destination action files.
+    """
+
+    source_metadata_file_path = os.path.join(
+        source_pack_base_path, source_metadata_file
+    )
+    dest_metadata_file_name = "%s.yaml" % (dest_action)
+    dest_metadata_file_path = os.path.join(
+        dest_pack_base_path, "actions", dest_metadata_file_name
+    )
+
+    if source_entry_point:
+        source_entry_point_file_path = os.path.join(
+            source_pack_base_path, "actions", source_entry_point
+        )
+        if str(source_entry_point).endswith("py"):
```

Comment:
Should be using the runner type in the metadata file to determine if this is python action.